### PR TITLE
 *: add_date can return mysql.Time (#9830)

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 
cherry-pick #9830

### CODE REVIEW

File: util/math/math.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+func StrLenOfUint64Fast(x uint64) int {
+	for i := 1; ; i++ {
+		if x <= uintSizeTable[i] {
+			return i
+		}
+	}
+}
+
+// StrLenOfInt64Fast efficiently calculate the string character lengths of an int64 as input
+func StrLenOfInt64Fast(x int64) int {
```
Comment (lamxTyler) [inline util/math/math.go]: Seems that they are not used?

Comment (winoros) [reply util/math/math.go]: Yes, just copied the whole file

**Comment by qw4990** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: util/math/math.go (hunk 1 above)
Comment (zz-jason) [reply util/math/math.go]: is this function introduced by another pr, and that pr is not cherry-picked to release 2.1?

**Pull request merged**